ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/ConnectionFactoryProxy.java

```diff
@@ -85,4 +86,24 @@ public Connection createConnection(String userName, String password) throws JMSE
         return connection;
     }
 
+    @Override
+    public JMSContext createContext() {
+        return xaConnectionFactory.createXAContext();
+    }
```

Comment:
I think we have to register the XAResource just like we do in ```ConnectionProxy```

```
XAJMSContext xaContext = xaConnectionFactory.createXAContext();
JMSContext context = new JMSContextProxy(xaContext, transactionHelper);
if (checkJTATransactionIsActive) {
     regiterXAResource(xaContext.getXAResource();
}
return context;
```

Also I think we need to provide ```JMSContextProxy``` to wrap ```JMSContext``` which delist the XAResource in ```close()``` method just like ```SessionProxy```.